sklearn_numba_dpex/kmeans/engine.py

```diff
@@ -105,16 +106,25 @@ def __init__(self, estimator):
             )
         self._is_in_testing_mode = _is_in_testing_mode == "1"
 
+    def accepts(self, X, y, sample_weight):
+
+        if (algorithm := self.estimator.algorithm) not in ("lloyd", "auto", "full"):
+            if self._is_in_testing_mode:
```

Comment:
As discussed in the bi-weekly plugin meeting, we can remove this condition and make the accepts method behave similarly in tests than in regular execution environments.